### DISCUSSION TITLE
Accuracy of Isochrone and Route Analysis Using Public Transit

### DISCUSSION THREAD

**Discussion by pipo3944** [valhalla/valhalla#4768] [Q&A]
I am conducting an isochrone analysis using public transit, but the results do not accurately reflect the use of public transit.

I am considering changing parameters during the request or build process to improve the results, but I am unsure what adjustments to make. I would appreciate your advice.

Here are the specific issues I am encountering:

**Start point**: 34.8706384, 138.3186816
**Mode of transport**: multimodal
**Travel time**: 30 minutes
**Start time**: 12:55
![スクリーンショット 2024-06-14 12 56 15 2](https://github.com/valhalla/valhalla/assets/12875256/d29aaf94-c8c6-4943-90b9-4783973b8052)

The map includes bus routes operating in the area, with GTFS data loaded into Valhalla.

At the start point, there is a bus stop within a 1-minute walk. A bus departs from this stop at 12:59, traveling for 16 minutes and arriving at the following location:
dep: 12:59, arr: 13:15
![スクリーンショット 2024-06-14 12 56 15 3](https://github.com/valhalla/valhalla/assets/12875256/74f39e18-59c2-42aa-bf98-1a9a05f3d31b)

Therefore, starting at 12:55, getting on the bus, and arriving at the mentioned location at 13:15 should leave about 10 minutes of travel time remaining within the 30-minute window.

However, the isochrone result shows that the area is not reachable within the given time frame.

I checked the route from the start point to the above-mentioned bus stop and the response indicated a total travel time of over an hour, without boarding the nearby bus.

The route response was as follows:
<details><summary>results</summary>
<pre>
{
  "trip": {
    "locations": [
      {
        "type": "break",
        "lat": 34.870638,
        "lon": 138.318681,
        "date_time": "2024-06-05T12:55",
        "time_zone_offset": "+09:00",
        "time_zone_name": "Asia/Tokyo",
        "original_index": 0
      },
      {
        "type": "break",
        "lat": 34.890475,
        "lon": 138.310055,
        "date_time": "2024-06-05T14:09",
        "time_zone_offset": "+09:00",
        "time_zone_name": "Asia/Tokyo",
        "side_of_street": "left",
        "original_index": 1
      }
    ],
    "legs": [
      {
        "maneuvers": [
          {
            "type": 1,
            "instruction": "Walk west on 焼津藤枝線/30.",
            "verbal_succinct_transition_instruction": "Walk west.",
            "verbal_pre_transition_instruction": "Walk west on 焼津藤枝線, 30.",
            "verbal_post_transition_instruction": "Continue for 600 meters.",
            "street_names": [
              "焼津藤枝線",
              "30"
            ],
            "time": 406.289,
            "length": 0.564,
            "cost": 598.054,
            "begin_shape_index": 0,
            "end_shape_index": 28,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 10,
            "instruction": "Turn right onto 焼津岡部線/213.",
            "verbal_transition_alert_instruction": "Turn right onto 焼津岡部線.",
            "verbal_succinct_transition_instruction": "Turn right.",
            "verbal_pre_transition_instruction": "Turn right onto 焼津岡部線, 213.",
            "verbal_post_transition_instruction": "Continue for 600 meters.",
            "street_names": [
              "焼津岡部線",
              "213"
            ],
            "time": 440.764,
            "length": 0.622,
            "cost": 660.647,
            "begin_shape_index": 28,
            "end_shape_index": 52,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 10,
            "instruction": "Turn right.",
            "verbal_transition_alert_instruction": "Turn right.",
            "verbal_succinct_transition_instruction": "Turn right.",
            "verbal_pre_transition_instruction": "Turn right.",
            "verbal_post_transition_instruction": "Continue for 300 meters.",
            "time": 218.411,
            "length": 0.307,
            "cost": 332.117,
            "begin_shape_index": 52,
            "end_shape_index": 56,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 15,
            "instruction": "Turn left.",
            "verbal_transition_alert_instruction": "Turn left.",
            "verbal_succinct_transition_instruction": "Turn left. Then Turn right onto 150.",
            "verbal_pre_transition_instruction": "Turn left. Then Turn right onto 150.",
            "verbal_post_transition_instruction": "Continue for less than 10 meters.",
            "time": 5.647,
            "length": 0.008,
            "cost": 8.47,
            "begin_shape_index": 56,
            "end_shape_index": 57,
            "verbal_multi_cue": true,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 10,
            "instruction": "Turn right onto 150/国道150号.",
            "verbal_transition_alert_instruction": "Turn right onto 150.",
            "verbal_succinct_transition_instruction": "Turn right.",
            "verbal_pre_transition_instruction": "Turn right onto 150, 国道150号.",
            "verbal_post_transition_instruction": "Continue for 300 meters.",
            "street_names": [
              "150",
              "国道150号"
            ],
            "time": 201.882,
            "length": 0.286,
            "cost": 307.823,
            "begin_shape_index": 57,
            "end_shape_index": 61,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 15,
            "instruction": "Turn left.",
            "verbal_transition_alert_instruction": "Turn left.",
            "verbal_succinct_transition_instruction": "Turn left.",
            "verbal_pre_transition_instruction": "Turn left.",
            "verbal_post_transition_instruction": "Continue for 400 meters.",
            "time": 269.176,
            "length": 0.37,
            "cost": 404.764,
            "begin_shape_index": 61,
            "end_shape_index": 71,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 30,
            "instruction": "Take the 焼津循環線　ゆりかもめ（日本坂PA～林叟院入口　フリー降車区間） toward 焼津駅前. (1 stop)",
            "verbal_pre_transition_instruction": "Take the 焼津循環線　ゆりかもめ（日本坂PA～林叟院入口　フリー降車区間） toward 焼津駅前.",
            "verbal_post_transition_instruction": "Travel 1 stop.",
            "time": 1930.999,
            "length": 0.305,
            "cost": 2719.725,
            "begin_shape_index": 71,
            "end_shape_index": 80,
            "depart_instruction": "Depart: 13:53 from 東益津小学校前.",
            "verbal_depart_instruction": "Depart at 13:53 from 東益津小学校前.",
            "arrive_instruction": "Arrive: 13:54 at 日本坂PA.",
            "verbal_arrive_instruction": "Arrive at 13:54 at 日本坂PA.",
            "transit_info": {
              "onestop_id": "Yaizushi-GTFS-JP_101004",
              "long_name": "焼津循環線　ゆりかもめ（日本坂PA～林叟院入口　フリー降車区間）",
              "headsign": "焼津駅前",
              "color": 28864,
              "text_color": 16777215,
              "operator_onestop_id": "Yaizushi-GTFS-JP_1000020222127",
              "operator_name": "焼津市",
              "operator_url": "http://www.city.yaizu.lg.jp/g06-002/kurashi/koutsuu/bus/index.html",
              "transit_stops": [
                {
                  "type": "stop",
                  "onestop_id": "Yaizushi-GTFS-JP_142_01_transit_station",
                  "name": "東益津小学校前",
                  "departure_date_time": "2024-06-05T13:53+09:00",
                  "assumed_schedule": true,
                  "lat": 34.886547,
                  "lon": 138.321294
                },
                {
                  "type": "stop",
                  "onestop_id": "Yaizushi-GTFS-JP_143_01_transit_station",
                  "name": "日本坂PA",
                  "arrival_date_time": "2024-06-05T13:54+09:00",
                  "assumed_schedule": true,
                  "lat": 34.889258,
                  "lon": 138.321841
                }
              ]
            },
            "travel_mode": "transit",
            "travel_type": "bus"
          },
          {
            "type": 36,
            "instruction": "Walk south.",
            "verbal_succinct_transition_instruction": "Walk south.",
            "verbal_pre_transition_instruction": "Walk south.",
            "verbal_post_transition_instruction": "Continue for 100 meters.",
            "time": 81.47,
            "length": 0.114,
            "cost": 126.705,
            "begin_shape_index": 80,
            "end_shape_index": 83,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 10,
            "instruction": "Turn right.",
            "verbal_transition_alert_instruction": "Turn right.",
            "verbal_succinct_transition_instruction": "Turn right.",
            "verbal_pre_transition_instruction": "Turn right.",
            "verbal_post_transition_instruction": "Continue for 400 meters.",
            "time": 286.882,
            "length": 0.405,
            "cost": 429.823,
            "begin_shape_index": 83,
            "end_shape_index": 92,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 23,
            "instruction": "Keep right at the fork.",
            "verbal_transition_alert_instruction": "Keep right at the fork.",
            "verbal_pre_transition_instruction": "Keep right at the fork.",
            "verbal_post_transition_instruction": "Continue for 200 meters.",
            "time": 129.176,
            "length": 0.183,
            "cost": 193.765,
            "begin_shape_index": 92,
            "end_shape_index": 103,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 23,
            "instruction": "Keep right at the fork.",
            "verbal_transition_alert_instruction": "Keep right at the fork.",
            "verbal_pre_transition_instruction": "Keep right at the fork.",
            "verbal_post_transition_instruction": "Continue for 500 meters.",
            "time": 373.882,
            "length": 0.524,
            "cost": 558.823,
            "begin_shape_index": 103,
            "end_shape_index": 110,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 10,
            "instruction": "Turn right.",
            "verbal_transition_alert_instruction": "Turn right.",
            "verbal_succinct_transition_instruction": "Turn right.",
            "verbal_pre_transition_instruction": "Turn right.",
            "verbal_post_transition_instruction": "Continue for 80 meters.",
            "time": 58.949,
            "length": 0.083,
            "cost": 111.89,
            "begin_shape_index": 110,
            "end_shape_index": 111,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          },
          {
            "type": 6,
            "instruction": "Your destination is on the left.",
            "verbal_transition_alert_instruction": "Your destination will be on the left.",
            "verbal_pre_transition_instruction": "Your destination is on the left.",
            "time": 0.0,
            "length": 0.0,
            "cost": 0.0,
            "begin_shape_index": 111,
            "end_shape_index": 111,
            "travel_mode": "pedestrian",
            "travel_type": "foot"
          }
        ],
        "summary": {
          "has_time_restrictions": false,
          "has_toll": false,
          "has_highway": false,
          "has_ferry": false,
          "min_lat": 34.870535,
          "min_lon": 138.309961,
          "max_lat": 34.890445,
          "max_lon": 138.321894,
          "time": 4452.238,
          "length": 3.842,
          "cost": 6452.611
        },
        "shape": "euioaA{thyfGhBxJz@~IP|HEdIa@nGw@bG]lD]jDeI`XaA~CgLd_@{CrIoCtEkDtDuEvDkFhD}ZxMwAl@ye@tQkAb@sBt@_OpFyLnE{Bx@yB`BoRtNaIbG{@n@{A_@ug@mMqA]c}@aUyBg@aj@aMoImBkCq@q|Am`@q@QmA[oOyDeGyAgC[eAMwCG_DDmD\\kB`@cBf@gCbAoBdAkBfAiCnB{HyKqZmb@w}@_rA_QuUaCx@yCuG_RqYgw@uiAmZ_b@wChB}DvAaB^yC?oEG{Fm@o|@iI{{@oJ_j@yEiEH_YyE]lD????mhDea@????LiBbBRzRnBrSpBpU|BeRbzCwDlm@o@lRCbP\\zILlHk@dH_@zAk@~BmBjDoYr_@u@bDOjBI~@G~F^tGtAxEtG~SxArHDvBHpEqGr`AqHbnA_C~_@cAfPwGfmAiOlhAsl@kM"
      }
    ],
    "summary": {
      "has_time_restrictions": false,
      "has_toll": false,
      "has_highway": false,
      "has_ferry": false,
      "min_lat": 34.870535,
      "min_lon": 138.309961,
      "max_lat": 34.890445,
      "max_lon": 138.321894,
      "time": 4452.238,
      "length": 3.842,
      "cost": 6452.611
    },
    "status_message": "Found route between points",
    "status": 0,
    "units": "kilometers",
    "language": "en-US"
  }
}
</pre>
</details> 

Visualizing the trajectory, it looked like this:
<img width="1345" alt="スクリーンショット 2024-06-17 12 18 39 2" src="https://github.com/valhalla/valhalla/assets/12875256/7c5f4450-4fda-4f36-a12e-f1c75ae4c1a5">

The route involved taking a bus for one stop, then walking further south, before continuing to the destination. This suggests the bus was not properly utilized. I tried adjusting the start time earlier and extending the travel time to 40 minutes, but the bus transit was still not used correctly.

What approach should I take to improve the results?

Thank you for your assistance.

**Comment by chrstnbwnkl** [comment]
The first thing I would do is use the `/expansion` endpoint to look at what edges the algorithm considered. It looks like it may not have used any transit here and just resorted to walking, in which case you would expect that during the expansion, no or few transit edges were considered.
